pycontrails/core/models.py

```diff
@@ -731,25 +762,271 @@ def interpolate_met(
     KeyError
         Parameter ``met_key`` not found in ``met``.
     """
-    if vector_key is None:
-        vector_key = met_key
+    vector_key = vector_key or met_key
 
     if (out := vector.get(vector_key, None)) is not None:
         return out
 
     if met is None:
-        raise ValueError(f"No variable key {vector_key} in ``vector`` and ``met`` is None")
+        raise ValueError(f"No variable key '{vector_key}' in 'vector' and 'met' is None")
+
+    if met_key in ("q", "specific_humidity"):
```

Comment:
i'd say `if met_key in ("q", "specific_humidity") and interp_kwargs["q_method"] is not None:`

Reply:
c190074612edb81b7716e1c83bc6034f97652be9